spring-context/src/test/java/org/springframework/context/groovy/GroovyBeanDefinitionReaderTests.java

```diff
@@ -0,0 +1,1299 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.context.groovy;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import groovy.lang.GroovyShell;
+import org.junit.jupiter.api.Test;
+
+import org.springframework.aop.SpringProxy;
+import org.springframework.beans.factory.ObjectFactory;
+import org.springframework.beans.factory.config.Scope;
+import org.springframework.beans.factory.groovy.GroovyBeanDefinitionReader;
+import org.springframework.context.support.GenericApplicationContext;
+import org.springframework.context.support.GenericGroovyApplicationContext;
+import org.springframework.core.io.ByteArrayResource;
+import org.springframework.stereotype.Component;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+
+public class GroovyBeanDefinitionReaderTests {
+
+	@Test
+	void importSpringXml() {
+		var appCtx = new GenericApplicationContext();
+		var reader = new GroovyBeanDefinitionReader(appCtx);
+
+		reader.loadBeanDefinitions(new ByteArrayResource(("""
+			beans {
+				importBeans "classpath:org/springframework/context/groovy/test.xml"
+			}
+			""").getBytes()));
+
+		appCtx.refresh();
+
+		var foo = appCtx.getBean("foo");
+		assertThat(foo).isEqualTo("hello");
+	}
+
+	@Test
+	void importBeansFromGroovy() {
+		var appCtx = new GenericApplicationContext();
+		var reader = new GroovyBeanDefinitionReader(appCtx);
+
+		reader.loadBeanDefinitions(new ByteArrayResource(("""
+			beans {
+				importBeans "classpath:org/springframework/context/groovy/applicationContext.groovy"
+			}
+			""").getBytes()));
+
+		appCtx.refresh();
+
+		var foo = appCtx.getBean("foo");
+		assertThat(foo).isEqualTo("hello");
+	}
+
+	@Test
+	void singletonPropertyOnBeanDefinition() {
+		var appCtx = new GenericApplicationContext();
+		var reader = new GroovyBeanDefinitionReader(appCtx);
+		reader.loadBeanDefinitions(new ByteArrayResource(("""
+		package org.springframework.context.groovy;
+		beans {
+			singletonBean(Bean1) { bean ->
+				bean.singleton = true
+			}
+			nonSingletonBean(Bean1) { bean ->
+				bean.singleton = false
+			}
+			unSpecifiedScopeBean(Bean1)
+		}
+		""").getBytes()));
+		appCtx.refresh();
+
+		assertThat(appCtx.isSingleton("singletonBean")).isTrue().as("singletonBean should have been a singleton");
```

Comment:
```suggestion
		assertThat(appCtx.isSingleton("singletonBean")).as("singletonBean should have been a singleton").isTrue();
```

In AssertJ, `.as(...)` must precede the (potentially failing) assertion method such as `.isTrue()`.

Please apply this change wherever applicable.